src/main/java/seedu/pennywise/model/ModelManager.java

```diff
@@ -189,6 +187,9 @@ public void updateFilteredIncomeList(Predicate<Entry> predicate) {
      * @return ObservableList of income pie chart data
      */
     public ObservableList<PieChart.Data> getIncomePieChartData() {
+        // make sure pie chart shows all unfiltered data
+        updateFilteredIncomeList(PREDICATE_SHOW_ALL_ENTRIES);
```

Comment:
This is added mainly for managing the manual switching behaviour. Currently manual switching calls `updateGraph` with `graphConfiguration` indicating a pie chart. I could either create a `updateFilteredXXList` method in `Logic` to update predicate and call this method in `updateGraph` when `graphType` is pie chart, or do the above, always clear filter before getting the pie chart data.

I followed the second one because I feel it is more guaranteed that pennywise follows our current specification: pie chart always shows all entries.

Reply:
lgtm